From d2acb4207113e53d38073812fac2b178e07ffeae Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Mon, 15 Jun 2015 12:43:47 +0800
Subject: [PATCH] MLK-11110: cpufreq: imx7-cpufreq: fix system resume failed

System resume failed easily after commit b993956109856d3, the commit is no
problem, but it exposes the potential deadlock bug which not unlock mutex if
cpufreq not change, remove these code since the cpufreq framwork has consider
it.

Signed-off-by: Robin Gong <b38343@freescale.com>
(cherry picked from commit d791bf25e07023f7a936fcabe6f641ad391048b3)
---
 drivers/cpufreq/imx7-cpufreq.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/cpufreq/imx7-cpufreq.c b/drivers/cpufreq/imx7-cpufreq.c
index 47a460d0..28d4cdd 100644
--- a/drivers/cpufreq/imx7-cpufreq.c
+++ b/drivers/cpufreq/imx7-cpufreq.c
@@ -41,9 +41,6 @@ static int imx7d_set_target(struct cpufreq_policy *policy, unsigned int index)
 	freq_hz = new_freq * 1000;
 	old_freq = clk_get_rate(arm_clk) / 1000;
 
-	if (new_freq == old_freq)
-		return 0;
-
 	rcu_read_lock();
 	opp = dev_pm_opp_find_freq_ceil(cpu_dev, &freq_hz);
 	if (IS_ERR(opp)) {
-- 
1.8.0

